Tests/FeaturevisorSDKTests/InstanceTests.swift

```diff
@@ -5,6 +5,137 @@ import XCTest
 
 class FeaturevisorInstanceTests: XCTestCase {
 
+    func testEncodingEvaluationWithNilValuesReturnsValidDatafileContent() throws {
+
+        //GIVEN
+        let evaluation = Evaluation(
+            featureKey: "feature123",
+            reason: .allocated
+        )
+
+        let expectedDictionary: [String: Any] = [
+            "featureKey": "feature123",
+            "reason": "allocated",
+        ]
+
+        let encoder = JSONEncoder()
+        let jsonData = try encoder.encode(evaluation)
+
+        //WHEN
+        guard
+            let decodedDictionary = try JSONSerialization.jsonObject(with: jsonData, options: [])
+                as? [String: Any]
+        else {
+            XCTFail("Failed to decode JSON to a dictionary.")
+            return
+        }
+        //THEN
+        XCTAssertEqual(decodedDictionary as NSDictionary, expectedDictionary as NSDictionary)
+
+    }
+
+    func testEncodeEvaluationReturnsValidDatafileContent() throws {
+
+        //GIVEN
+        let traffic = Traffic(
+            key: "key",
+            segments: .plain("segment"),
+            percentage: 13,
+            allocation: []
+        )
+        let overrideFeature = OverrideFeature(
+            enabled: false,
+            variation: "variation",
+            variables: [:]
+        )
+        let variation = Variation(
+            description: "description",
+            value: "value",
+            weight: 3,
+            variables: []
+        )
+        let variableSchema = VariableSchema(
+            key: "key",
+            type: .object,
+            defaultValue: .object(["": .boolean(false)])
+        )
+
+        let evaluation = Evaluation(
+            featureKey: "feature123",
+            reason: .allocated,
+            bucketValue: 42,
+            ruleKey: "rule456",
+            enabled: true,
+            traffic: traffic,
+            sticky: overrideFeature,
+            initial: overrideFeature,
+            variation: variation,
+            variationValue: "value",
+            variableKey: "color",
+            variableValue: .string(""),
+            variableSchema: variableSchema
+        )
+
+        let mockedTraffic: [String: Any] = [
```

Comment:
Don't use mocked as it's confusing. Rather `expectedTraffic`. Also move expected stuff under "THEN"